### ISSUE TITLE
ensureDir hangs when folder is readonly

### ISSUE THREAD

**Issue by tdtc** [jprichardson/node-fs-extra#528]
`ensureDir` function doesn't response callback when folder is in readonly mode (I use ubuntu). I don't know how to debug this problems.  Please help me resolve!

**Comment by RyanZim** [comment]
It errors out with `EACCES: permission denied, mkdir '/app/t/c'` for me: https://runkit.com/57ebc041d4a27f140028fd78/5a33f49ef88d280012b07780

What version of fs-extra are you using?

**Comment by tdtc** [comment]
Sorry for my late reply. 
I think it is problem with kind of partition that I'm using. I'm using mergerfs (mergerfs logically merges multiple paths together https://github.com/trapexit/mergerfs) to mount file system. I have just tested in non-mergerfs partition (partition is mounted with readonly) everything is ok to catch error. 

log message when using mkdir command:
 
- With non-mergerfs partition is mounted readonly: "cannot create directory '/archive/1712/2106': Read-only file system"
- With mergerfs: "cannot create directory '/archive/hugedisk/1206': No such file or directory". I think mergerfs try to find partitions to write but not found, so it throw error "No such file or directory". I will contact with author of mergerf to make clearly. 

How fs-extra deal with kind of error message "No such file or directory". I also tested with the same tool: mkdirp, make-dir and get the same result like fs-extra.
